### PR TITLE
ci: add PR tittle formatting check

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description
This PR introduces a title formatting check for all pull requests to maintain a consistent style across the project.

## Motivation and Context
There was a need to enforce a consistent title formatting for all PRs to improve readability and maintainability. This change will ensure that all PR titles adhere to a predefined format, making them easier to understand and manage.

## Changes
- A new PR title checker workflow has been added, which will run on every PR event such as opened, edited, synchronized, labeled, and unlabeled.
- A new configuration file, pr-title-checker-config.json, has been created to define the rules of the title formatting.
- Updated the version of the `github-tag-action` used in `github-tag-and-release.yml` from v6.1 to v6.2.

## Fixes Jira Issue

[https://jira.esss.lu.se/browse/SWAP-3868](https://jira.esss.lu.se/browse/SWAP-3868)

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: .github/pr-title-checker-config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "LABEL": {
+    "name": ""
+  },
+  "CHECKS": {
+    "NOTE": "You can test the regex here: https://regex101.com/r/W2f7Yr/1",
+    "regexp": "((?<type>build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test|BREAKING CHANGE)(?<scope>\\(\\w+\\))?(?<breaking_change>!?): (?<summary>[\\w -]+))(?<body>[\\w\\s-]+)(?<footer>(?<=\\v\\v)(?<footer_token>[\\w-]+): (?<footer_value>[\\w -]+)|$)",
```
Comment (janosbabik) [inline .github/pr-title-checker-config.json:7]: Can we simplify the regex? Currently, it allows only words in the body. What are your thoughts on using this instead?
`((?<type>build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test|BREAKING CHANGE)(?<scope>\(\w+\))?(?<breaking_change>!?): (?<body>.+)$)`

Comment (martin-trajanovski) [reply .github/pr-title-checker-config.json:7]: Sure maybe this is a good idea.

**Review by janosbabik — APPROVED** [review]

**Review by simonfernandes — COMMENTED** [review]
Looks like a really nice feature! I have a few questions/comments:

- Do we want to enforce a lowercase first letter? e.g. "fix: **p**roposal cloning"
- Similarly do we also want to enforce no full stop at the end?
- Does this need to allow `*` for scope, e.g. "fix(*) proposal cloning"?

I have made the assumption that we're following the [angular commit convention](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md) that we link to in our readme.

**Comment by martin-trajanovski** [comment]
> Looks like a really nice feature! I have a few questions/comments:
> 
> * Do we want to enforce a lowercase first letter? e.g. "fix: **p**roposal cloning"
> * Similarly do we also want to enforce no full stop at the end?
> * Does this need to allow `*` for scope, e.g. "fix(*) proposal cloning"?
> 
> I have made the assumption that we're following the [angular commit convention](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md) that we link to in our readme.

Those are quite good points. I will try to go through the angular commit convention and fix the regex properly.

**Review by simonfernandes — COMMENTED** [review]
The full stop regex works with a single word body but not a multi word body, maybe it needs to allow whitespace? `(?<body>([a-z|\s]).+[^.])$)`

**Comment by simonfernandes** [comment]
@martin-trajanovski Actually what I said isn't correct anyway, sorry. I was saying that it didn't seem to prevent a full stop if the body had multiple words like this example:

image

But I just noticed that there is space after the full stop which is why it's allowed.

So the regex correctly prevents this: `build(Breaking): la asdf asdf asdf.`
But doesn't prevent this: `build(Breaking): la asdf asdf asdf.  ` (note the space at the end)

**Comment by martin-trajanovski** [comment]
> @martin-trajanovski Actually what I said isn't correct anyway, sorry. I was saying that it didn't seem to prevent a full stop if the body had multiple words like this example:
> 
> image
> 
> But I just noticed that there is space after the full stop which is why it's allowed.
> 
> So the regex correctly prevents this: `build(Breaking): la asdf asdf asdf.` But doesn't prevent this: `build(Breaking): la asdf asdf asdf. ` (note the space at the end)

Should be fixed now. Thanks for the good feedback.

**Review by simonfernandes — APPROVED** [review]

**Pull request merged**